modules/touch/src/TouchInteraction.cpp

```diff
@@ -0,0 +1,779 @@
+#include <openspace/scene/scene.h>
+#include <openspace/util/time.h>
+#include <openspace/util/keys.h>
+#include <ghoul/misc/invariants.h>
+#include <ghoul/logging/logmanager.h>
+
+#include <glm/gtx/quaternion.hpp>
+
+#ifdef OPENSPACE_MODULE_GLOBEBROWSING_ENABLED
+#include <modules/globebrowsing/globes/renderableglobe.h>
+#include <modules/globebrowsing/globes/chunkedlodglobe.h>
+#include <modules/globebrowsing/geometry/geodetic2.h>
+#endif
+
+#include <cmath>
+#include <functional>
+#include <fstream>
+#include <glm/ext.hpp>
+
+namespace {
+    const std::string _loggerCat = "TouchInteraction";
+}
+
+using namespace TUIO;
+namespace openspace {
+
+TouchInteraction::TouchInteraction()
+    : properties::PropertyOwner("TouchInteraction")
+    , _origin("origin", "Origin", "")
+    , _unitTest("Click to take a unit test", "Take a unit test saving the LM data into file", false)
+    , _touchActive("TouchEvents", "True if we have a touch event", false, properties::Property::Visibility::Hidden)
+    , _reset("Default Values", "Reset all properties to default", false)
+    , _maxTapTime("Max Tap Time", "Max tap delay (in ms) for double tap", 300, 10, 1000)
+    , _deceleratesPerSecond("Decelerates per second", "Number of times velocity is decelerated per second", 240, 60, 300)
+    , _touchScreenSize("TouchScreenSize", "Touch Screen size in inches", 55.0f, 5.5f, 150.0f)
+    , _tapZoomFactor("Tap zoom factor","Scaling distance travelled on tap", 0.2, 0.0, 0.5)
+    , _nodeRadiusThreshold("Activate direct-manipulation", "Radius a planet has to have to activate direct-manipulation", 0.2f, 0.0f, 1.0f)
+    , _rollAngleThreshold("Interpret roll", "Threshold for min angle for roll interpret", 0.025f, 0.0f, 0.05f)
+    , _orbitSpeedThreshold("Activate orbit spinning", "Threshold to activate orbit spinning in direct-manipulation", 0.005f, 0.0f, 0.01f)
+    , _spinSensitivity("Sensitivity of spinning", "Sensitivity of spinning in direct-manipulation", 1.0f, 0, 2)
+    , _inputStillThreshold("Input still", "Threshold for interpreting input as still", 0.0005f, 0, 0.001)
+    , _centroidStillThreshold("Centroid stationary", "Threshold for stationary centroid", 0.0018f, 0, 0.01) // used to void wrongly interpreted roll interactions
+    , _interpretPan("Pan delta distance", "Delta distance between fingers allowed for interpreting pan interaction", 0.015f, 0, 0.1)
+    , _slerpTime("Time to slerp", "Time to slerp in seconds to new orientation with new node picking", 3, 0, 5)
+    , _guiButton("GUI Button", "GUI button size in pixels.", glm::ivec2(32, 64), glm::ivec2(8, 16), glm::ivec2(128, 256))
+    , _friction("Friction", "Friction for different interactions (orbit, zoom, roll, pan)", glm::vec4(0.01, 0.025, 0.02, 0.02), glm::vec4(0.0), glm::vec4(0.2))
+    
+    , _vel{ glm::dvec2(0.0), 0.0, 0.0, glm::dvec2(0.0) }
+    , _sensitivity{glm::dvec2(0.0808181818181818, 0.0454545454545455), 4.0, 2.75, glm::dvec2(0.0808181818181818, 0.0454545454545455) }
```

Comment:
Reason is to nullify sensitivity difference between X-axis and Y-axis on a 16:9 aspect ratio. Not the best solution as aspect ratio easily can change, it's not a super important feature for the user experience anyways. I'll change them to `0.08` and `0.045` for now.